torch/_functorch/aot_autograd.py

```diff
@@ -1347,7 +1400,37 @@ def merge_view_inputs(
             # Case where none of the aliases have a ._base
             # we generate a synthetic base without gradients, and generate views off of it
             example_idx = aliased_input_indices[0]
-            synthetic_base = torch.Tensor(fwd_inputs[example_idx].untyped_storage())
+            if is_runtime:
+                # Wondering when we hit this case? a known example is when compiling
+                # the optimizer step, where the parameters came from an LSTM module.
+                # LSTM performs weight packing, so weights are views from a shared buffer.
+                # NOTE: There is one case where this is unsafe:
+                # torch.Tensor(storage) will ALWAYS create a 1D tensor, which is not necessarily
+                # the same shape as the "actual" base that the tensor came from.
+                # For the most part this is fine, because we always use as_strided()
+                # to generate the original aliased inputs again.
+                # If we were to use view-replay though, this could cause the aliased views
+                # to have incorrect sizes.
+                example_alias = fwd_inputs[example_idx]
+                # We don't actually have a convenient way of going from storage -> tensor,
+                # So using set_() here (we suffer some minor overhead, but this case is rare).
+                synthetic_base = torch.empty((), dtype=example_alias.dtype, device=example_alias.device)
+                synthetic_base.set_(example_alias.untyped_storage())
```

Comment:
this is a behavior change from before, I guess?

Reply:
The "old behavior" was just that this case was broken and would error - "this case" being "inputs share storage, but none of them have a `._base` attribute (it would error because my original code tried to construct `torch.Tensor(storage)` with a cuda storage, which is not allowed).

I spent some time looking at the LSTM code and actually couldn't really find the logic that set things up so that there was no ._base attribute on the aliased tensors. So locally, I have a new test that just simulates this behavior by calling .detach() on the views before passing them in (going finish going through PR feedback before pushing more changes).